internal/controller/serverset/serverset.go

```diff
@@ -286,31 +292,100 @@ func (c *external) reconcileVolumesFromTemplate(ctx context.Context, cr *v1alpha
 		}
 
 		if deleteAndCreate {
-			if err := c.kube.Delete(ctx, &volumes[idx]); err != nil {
-				fmt.Printf("error deleting volume %v", err)
-				return err
-			}
-			err := kube.WaitForKubeResource(ctx, resourceReadyTimeout, kube.IsVolumeDeleted, c.kube, volumes[idx].Name, cr.Namespace)
+			res := &v1alpha1.VolumeList{}
+			err := kube.ListResourceFromLabelIndex(ctx, c.kube, "bootvolume", idx, res)
 			if err != nil {
 				return err
 			}
-			var createdVolume v1alpha1.Volume
-			if createdVolume, err = c.createBootVolume(ctx, cr, volumes[idx].Name); err != nil {
-				return err
+			if len(res.Items) > 1 {
+				return fmt.Errorf("found too many volumes for index %d ", idx)
 			}
-			gotServer, err := c.getServer(ctx, getNameFromIndex(cr.Name, "server", idx), cr.Namespace)
-			if err != nil {
-				return err
-			}
-			gotServer.Spec.ForProvider.VolumeCfg.VolumeID = meta.GetExternalName(&createdVolume)
-			err = c.kube.Update(ctx, gotServer)
-			if err != nil {
-				return err
-			}
-			// wait for server to become ready again after re-attaching volume
-			err = kube.WaitForKubeResource(ctx, resourceReadyTimeout, IsServerAvailable, c.kube, getNameFromIndex(cr.Name, "server", idx), cr.Namespace)
-			if err != nil {
-				return err
+			if len(res.Items) > 0 {
+				condemnedVolume := res.Items[0]
+				volumeVersion, _ := strconv.Atoi(condemnedVolume.Labels[fmt.Sprintf(serversetVersionLabel, "bootvolume")])
```

Comment:
the `ok` value should be checked, you never know who is accessing k8s and deleting/changing labels